packages/next/src/export/worker.ts

```diff
@@ -548,3 +548,17 @@ process.on('rejectionHandled', () => {
   // prefetching patterns to avoid waterfalls. We ignore logging these.
   // We should've already errored in anyway unhandledRejection.
 })
+
+const FATAL_UNHANDLED_NEXT_API_EXIT_CODE = 78
+
+process.on('uncaughtException', (err) => {
+  if (isDynamicUsageError(err)) {
+    console.error(
+      'A Next.js API that uses exceptions to signal framework behavior was uncaught. This suggests improper usage of a Next.js API. The original error is printed below and the build will now exit.'
+    )
+    console.error(err)
+    process.exit(FATAL_UNHANDLED_NEXT_API_EXIT_CODE)
```

Comment:
Is the exit code just for the tests? If so, checking for this log instead would be better than creating a new special exit code.

Reply:
My rationale is that if we get an error report with a status code having a fast path to know that it came from an unhandled exception of a framework API might be helpful. We could exit with any code but using 1 seems to lose specificity. Then again we could just use the log message